### PR TITLE
Generate Application Layout spec

### LINKED ISSUE #735: Generate ApplicationLayout spec
We generate `ApplicationLayout` for new apps, but not an associated spec.

Here's what it could look like, but probably without the  `it` block (shown for example), located at: `spec/web/views/application_layout_spec.rb`
```
require "spec_helper"
require_relative "../../../apps/web/views/application_layout"

describe Web::Views::ApplicationLayout do
  let(:layout)   { Web::Views::ApplicationLayout.new(template, {}) }
  let(:rendered) { layout.render }
  let(:template) { Hanami::View::Template.new("apps/web/templates/application.html.slim") }

  it "has #site_name" do
    layout.site_name.must_equal "My Hanami Web App"
    rendered.must_include "My Hanami Web App"
  end
end
```

Note: this is a suggestion for `v1.1`, **not** `v1.0` :)

### PR DESCRIPTION
**Pull request by akhramov**
This Pull Request

* Adds two templates (minitest and rspec) for application_layout_spec.rb
* Tweaks `new` and `generate` commands to generate the spec file as well
* Contains various test fixes and additions.
* Hopefully closes #735

*Note*: I wasn't sure where to place template files. Is there any better place?

### CODE REVIEW

File: spec/integration/cli/generate/app_spec.rb [hunk 1]
```diff
@@ -105,11 +109,17 @@
...
+
+          #
+          # spec/admin/views/application_layout_spec.rb
+          #
+          expect("spec/admin/views/application_layout_spec.rb").to have_file_content(%r{apps/admin/templates/application.html.haml})
         end
       end
     end # haml
 
+
```
Comment (AlfonsoUceda) [inline spec/integration/cli/generate/app_spec.rb]: extra line

Comment (akhramov) [inline spec/integration/cli/generate/app_spec.rb]: Thanks! Fixed.

**Comment by AlfonsoUceda** [comment]
@hanami/core this LGTM to me, what do you think?

**Comment by cllns** [comment]
great! I changed the `base` to `develop`

**Comment by mereghost** [comment]
Merging this, then!

**Pull request merged**